include/xtensor/xutils.hpp

```diff
@@ -23,6 +23,7 @@
 #include "xtl/xfunctional.hpp"
 #include "xtl/xtype_traits.hpp"
 
+#include "xshape.hpp"
```

Comment:
`xutils.hpp` should not depend on `xshape.hpp` (nor on any other file except `xtensor_config.hpp` since it is a really "low-level" file).

Forward declaration of dynamic shape should be used instead.

Reply:
`#include "shape.hpp"` is now still there, but xshape.hpp only contains a few forward declarations that would look weird inside xutils.